subprojects/soak/src/integTest/groovy/org/gradle/vfs/VirtualFileSystemRetentionSoakTest.groovy

```diff
@@ -96,17 +97,22 @@ class VirtualFileSystemRetentionSoakTest extends DaemonIntegrationSpec implement
         daemon.assertIdle()
 
         when:
-        NUMBER_OF_REPETITIONS.times { iteration ->
-            changeAllSourceFiles(iteration)
-            waitForChangesToBePickedUp()
+        numberOfChangeBatches.times { iteration ->
+            changeSourceFiles(iteration, numberOfChangedSourcesFilesPerBatch)
+            waitBetweenChangesToAvoidOverflow()
         }
         then:
         succeeds("assemble")
         daemons.daemon.logFile == daemon.logFile
         daemon.assertIdle()
         assertWatchingSucceeded()
-        receivedFileSystemEvents >= minimumExpectedFileSystemEvents(TOTAL_NUMBER_OF_SOURCES, NUMBER_OF_REPETITIONS)
-        retainedFilesInCurrentBuild - TOTAL_NUMBER_OF_SOURCES == retainedFilesSinceLastBuild
+        receivedFileSystemEvents >= minimumExpectedFileSystemEvents(numberOfChangedSourcesFilesPerBatch, numberOfChangeBatches)
+        retainedFilesInCurrentBuild - numberOfChangedSourcesFilesPerBatch == retainedFilesSinceLastBuild
+    }
+
+    private void waitBetweenChangesToAvoidOverflow() {
+        waitForChangesToBePickedUp()
+        waitForChangesToBePickedUp()
```

Comment:
FTR, let's make this explicit instead of the double call.

Reply:
Done.